lib/libxdp/libxdp.c

```diff
@@ -3223,6 +3223,43 @@ static int xdp_multiprog__unpin(struct xdp_multiprog *mp)
 	return err;
 }
 
+static void xdp_detach_links(uint32_t ifindex) {
+	struct bpf_link_info link_info;
+	u_int32_t link_info_len = sizeof(link_info), id;
+	int err, fd;
+
+	while (true) {
+		err = bpf_link_get_next_id(id, &id);
```

Comment:
The first time through the loop, `id` will be passed in uninitialised here.

Reply:
fixed it